### PR TITLE
Support create new cluster in spring apps maven plugin

### PR DESCRIPTION
**Pull request by Flanker32**
What does this implement/fix? Explain your changes.
---------------------------------------------------
…

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/config/AppDeploymentRawConfig.java [hunk 1]
```diff
@@ -19,4 +22,13 @@ public class AppDeploymentRawConfig {
     private String deploymentName;
     private String jvmOptions;
     private String runtimeVersion;
+
+    public void saveSpringCloudDeployment(@Nonnull final SpringCloudDeployment activeDeployment) {
```
Comment (wangmingliang-ms) [inline azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/config/AppDeploymentRawConfig.java:26]: rename to `from(...)`? see also `SpringCloudAppConfig/SpringCloudDeploymentConfig.from(xxx)`

File: azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/config/AppRawConfig.java [hunk 2]
```diff
@@ -5,14 +5,32 @@
...
 public class AppRawConfig {
     private String subscriptionId;
     private String resourceGroup; // optional
     private String clusterName;
     private String appName;
     private String isPublic;
+    private String region;
+    private String sku;
+    private String environment;
+    private String environmentResourceGroup;
```
Comment (wangmingliang-ms) [inline azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/config/AppRawConfig.java]: how about adding a `ClusterRawConfig`, as an alternative config to `clusterName`.

File: azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/config/SpringCloudAppConfig.java [hunk 3]
```diff
@@ -29,6 +31,10 @@
 public class SpringCloudAppConfig {
     private String subscriptionId;
     private String clusterName;
+    private String region;
+    private String sku;
+    private String environment;
+    private String environmentResourceGroup;
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-springcloud-lib/src/main/java/com/microsoft/azure/toolkit/lib/springcloud/config/SpringCloudAppConfig.java]: how about adding a `SpringCloudClusterConfig cluster`?

File: azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java [hunk 4]
```diff
@@ -154,8 +165,16 @@ protected void doExecute() throws AzureExecutionException {
             loginAzure();
             promptAndSelectSubscription();
 
-            selectAppCluster();
-            configCommon();
+            // prompt to select existing cluster or create a new one
+            useExistingCluster = this.wrapper.handleConfirm("Using existing cluster in Azure (Y/n):", true, true);
+            final SpringCloudCluster cluster = useExistingCluster ? selectAppCluster() : configCluster();
+            useExistingApp = !Objects.isNull(cluster) && !parentMode &&
+                this.wrapper.handleConfirm(String.format("Using existing app in cluster %s (Y/n):", cluster.getName()), true, true);
```
Comment (wangmingliang-ms) [inline azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java]: we will definitely create a new app if user choose to create a new cluster.

Comment (Flanker32) [reply azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java]: Yes, in this case `configureCluster()` will return null, so useExistingApp will be false in this case.

File: azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/config/AppDeploymentRawConfig.java (hunk 1 above)
Comment (Flanker32) [reply azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/config/AppDeploymentRawConfig.java:26]: I prefer to keep current naming as this method is to save remote deployment configuration in an existing `AppDeploymentRawConfig` object while `from(...)` is more likely to create a new instance

Comment (Flanker32) [reply azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/config/AppDeploymentRawConfig.java:26]: -> merge

File: azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java [hunk 5]
```diff
@@ -116,6 +124,9 @@ public class ConfigMojo extends AbstractMojoBase {
     @Parameter(property = "advancedOptions")
     private Boolean advancedOptions;
 
+    private boolean useExistingCluster = false;
```
Comment (Flanker32) [inline azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java:133]: remove this from mojo prarameter lists

File: azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java [hunk 6]
```diff
@@ -171,12 +190,76 @@ protected void doExecute() throws AzureExecutionException {
         }
     }
 
+    private SpringCloudCluster configCluster() throws IOException, InvalidConfigurationException {
+        configureClusterName();
+        configureResourceGroup();
+        final Region region = configureRegion();
+        final Sku sku = configureSku(region);
+        if (sku.isConsumptionTier()) {
```
Comment (Flanker32) [inline azure-spring-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java]: may comment it before consumption ga

**Pull request merged**